### PR TITLE
Add GettingStarted for Qiskit Nature

### LINKED ISSUE #119: Update Installation Instructions
When chemistry was part of Aqua, and Aqua installed by default with qiskit, PySCF was changed to an optional install since it would otherwise install for everyone and some people had problems with their installation and did not even want to use chemistry function.

Given that chemistry is part of nature here, and you still might want to install it for physics or other aspects, maybe leaving PySCF as optional is fine. On the other hand today its mostly chemistry and our tests and tutorials etc use this extensively. So I figured I'd raise this for discussion as to whether we want to go back to having it installed as a dependent, like we had in the past before making it optional for reasons given above, or leave it be as optional....

### PR DESCRIPTION
**Pull request by woodsp-ibm**
Add a GettingStarted documentation page for Qiskit Nature that 'augments' the main one of Qiskit by providing additional installation information etc specific to Qiskit Nature.

Closes #119

### CODE REVIEW

**Review by manoelmarques — APPROVED** [review]

File: docs/getting_started.rst
Comment (mrossinek) [inline docs/getting_started.rst:14]:
```suggestion
specific to Qiskit Nature.
```

File: docs/getting_started.rst
Comment (mrossinek) [inline docs/getting_started.rst:18]:
```suggestion
classical library/programs for molecular problems.
```

File: docs/getting_started.rst
Comment (mrossinek) [inline docs/getting_started.rst:26]:
```suggestion
    In your virtual environment where you installed Qiskit simply add Qiskit Nature to the
```

File: docs/getting_started.rst
Comment (mrossinek) [inline docs/getting_started.rst:50]:
```suggestion
   features of Qiskit, you should first follow Qiskit's `"Install from source"` instructions
```

File: docs/getting_started.rst
Comment (mrossinek) [inline docs/getting_started.rst:98]:
```suggestion
compute molecular information, such as one and two body integrals. This is needed as problem input to
```

File: docs/getting_started.rst
Comment (mrossinek) [inline docs/getting_started.rst:99]:
```suggestion
algorithms that compute properties of molecules, such as the ground state energy, so at least one such
```

File: docs/getting_started.rst
Comment (mrossinek) [inline docs/getting_started.rst:120]: Is this the only reason for all those custom templates added by this PR? Rendering the docs locally is really underwhelming to be honest..
I also did not see all of these extensions in Terra. Did you copy them from another Qiskit repo?
screenshot_1621233675

**Review by mrossinek — CHANGES_REQUESTED** [review]
Thanks @woodsp-ibm! Before merging, we should fix some typos.

@manoelmarques Is there a way to include `rst` files in our spell checking?

File: docs/getting_started.rst
Comment (manoelmarques) [reply docs/getting_started.rst:14]: The current spell check uses pylint which only looks at python source code. There is a PR on hold on Optimization to posssibly use sphinxcontrib.spelling which is called from sphinx and I believe would look at the whole documentation.

File: docs/getting_started.rst
Comment (woodsp-ibm) [reply docs/getting_started.rst]: I have the spellchecker turned on locally. I missed the above one - its right over a highlighted block on my machine so I did not notice the wiggly underlining. But as Manoel mentioned if we switch spellchecked to sphinx output it would take care of mistakes like that!

File: docs/getting_started.rst
Comment (woodsp-ibm) [reply docs/getting_started.rst:26]: No that is not what is wanted. where you did 'pip install qiskit` you do `pip install qiskit[nature]` i.e add nature to the list same as visualization example.  I guess its fine either way on qiskit capitalization but maybe I should have quoted nature to make it show up. I'll do that instead

File: docs/getting_started.rst
Comment (woodsp-ibm) [reply docs/getting_started.rst:120]: This follows the look/feel of the GettingStarted in the main Qiskit documentation and the extensions are defined in `qiskit` repo. Yes the module is a copy - I need to check with Paul concerning lack of copyright as I copied it as is and it does not have an.

**Review by mrossinek — APPROVED** [review]

**Pull request merged**
